csrc/multidevice/lower_communication.cpp

```diff
@@ -170,7 +170,7 @@ void lowerToGather(
     at::Tensor output_tensor,
     std::vector<std::shared_ptr<Communication>>& comms) {
   // we create as many 'Gathers' as there are devices in the receiver mesh
-  auto sender_mesh = input_tv->getDeviceMesh();
+  const auto& sender_mesh = input_tv->getDeviceMesh();
```

Comment:
```suggestion
  const DeviceMesh& sender_mesh = input_tv->getDeviceMesh();
```

Style nit: I tend to not use auto unless the type is clear from the lines around it. That leads to more typing but is general worthwhile for future readers (especially newcomers) to easily understand the code. https://google.github.io/styleguide/cppguide.html#Type_deduction